src/pages/resultsView/mutation/MutationMapperStore.ts

```diff
@@ -142,10 +143,33 @@ export class MutationMapperStore {
         }
     }, undefined);
 
+    readonly allTranscripts = remoteData<EnsemblTranscript[] | undefined>({
+        invoke: async()=>{
+            if (this.gene) {
+                return fetchEnsemblTranscriptsByEnsemblFilter({"hugoSymbols":[this.gene.hugoGeneSymbol]});
+            } else {
+                return undefined;
+            }
+        },
+        onError: (err: Error) => {
+            console.log("Getting all transcripts failed");
+        }
+    }, undefined);
+
     readonly canonicalTranscript = remoteData<EnsemblTranscript | undefined>({
+        await: () => [
+            this.allTranscripts
+        ],
         invoke: async()=>{
             if (this.gene) {
-                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource);
+                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource).catch(async() => {
```

Comment:
This seems to me a bit complicated as a return statement, maybe we should move the logic within the catch block into another utility function?

Reply:
is it a failure that we expect or a real break down?

Reply:
one that we expect. As a backup it uses the same service, so a real breakdown would still fail